pallets/staking/src/lib.rs

```diff
@@ -1998,6 +1998,16 @@ impl<T: Trait> Module<T> {
         true
     }
 
+    /// Return reward curve points
+    pub fn get_curve() -> Vec<(Perbill, Perbill)> {
+        let curve = &T::RewardCurve::get();
+        let mut points: Vec<(Perbill, Perbill)> = Vec::new();
```

Comment:
You can use `with_capacity()` for the vector to determine the exact length

Reply:
The number of points change depending on the parameters (as in the description). So we don't know the capacity in advance to be able to use `Vec::with_capacity()`.

Reply:
can't we have Vec::with_capacity(curve.points.len())